core/templates/dev/head/collection_editor/Services.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Standalone services for the exploration editor page.
```

Comment:
As discussed offline with @seanlip, we both agreed having 2 separate handlers makes sense. However, the handlers serve different purposes. First, there will be a single handler which retrieves all collection data needed by the collection viewer (as there currently is). Second, there will be a second data handler in the collection editor that provides any *additional* information not already provided by the first data handler. This additional data is specific to the editor.

In the frontend, there will need to be one data service for retrieving the collection data (this exists as part of the collection learner view work) and a second for retrieving the additional editor data (this probably doesn't need to be modeled in the same way as the current collection data service--we can discuss that if you'd like). Finally, both data services will need to be updated to return promise objects (I will be implementing this in the CollectionDataService today). Promise objects can be chained together, allowing the frontend to be initialized only after both promises have been satisfied (ie, we can request both the collection and its metadata to be loaded at the same time, but not end up with a partially loaded editor state).

Please let me know if this sounds good and is clear.